gpii/node_modules/preferencesServer/src/preferencesService.js

```diff
@@ -73,12 +124,270 @@ var fluid = fluid || require("infusion");
                 listener: "gpii.preferencesServer.preferencesService.updateGpiiKey",
                 args: ["{dataStore}.updateGpiiKey", "{arguments}.0"],
                 namespace: "updateGpiiKey"
+            }],
+            onCloudSafeCredCreate: [{
+                func: "{dataStore}.findById",
+                args: ["{arguments}.0"],
+                namespace: "findById"
+            }, {
+                func: "gpii.dbOperation.verifyExists",
+                args: ["{arguments}.0", "prefsSafe", gpii.preferencesServer.errors.missingPrefsSafe.message,
+                    gpii.preferencesServer.errors.missingPrefsSafe.errorCode]
+            }, {
+                func: "{expressUserUtils}.createNewUser",
+                args: [{username: "{arguments}.1.username", password: "{arguments}.1.password"}],
+                namespace: "createNewUser"
+            }, {
+                funcName: "gpii.dbOperation.dbDataStore.addRecord",
+                args: ["{dbDataStore}.saveDataSource",
+                       "gpiiCloudSafeCredential",
+                       "id", {
+                           type: "gpiiCloudSafeCredential",
+                           schemaVersion: "0.1",
+                           prefsSafeId: "{arguments}.1.prefsSafeId",
+                           gpiiExpressUserId: "{arguments}.0._id"
+                       }],
+                namespace: "addRecord"
+            }, {
+                funcName: "{dataStore}.findById",
+                args: ["{arguments}.0.id"]
+            }],
+            onCloudSafeUnlock: [{
+                func: "{expressUserUtils}.unlockUser",
+                args: ["{arguments}.0.username", "{arguments}.0.password"], // username, password
+                namespace: "unlockUser"
+            }, {
+                func: "{dataStore}.findSafeByExpressUserLookup",
+                args: ["{arguments}.0._id"],
+                namespace: "findSafeByExpressUserLookup"
+            }, {
+                func: "{dataStore}.findById",
+                args: ["{arguments}.0.prefsSafeId"],
+                namespace: "findById"
             }]
         }
     });
 
     // APIs for Preferences and Preferences Safes
 
+    gpii.preferencesServer.preferencesService.errors = fluid.freezeRecursive({
+        gpiiKeyMissing: "GPII key \"%gpiiKey\" does not exist",
+        gpiiKeyExisted: "GPII key \"%gpiiKey\" already exists",
+        noUpdateOnSnapset: "Cannot update:  GPII key \"%gpiiKey\" is a snapset"
+    });
+
+    /**
+     * Returns a preferences safe using it's id. If the safe is not found an error object
+     * is returned.
+     *
+     * @param  {Object} dataStore - Instance of `gpii.dbOperation.dataStore`
+     * @param  {String} prefsSafeId Prefs Safe ID
```

Comment:
Fix the jsdoc format at line 184, 185. also remove the extra space in btw "@param" and "{".

Reply:
updated